docs/pipeline/triggering/DRIFT_TRIGGERS.md

```diff
@@ -0,0 +1,231 @@
+
+### Main Architecture
+
+```mermaid
+classDiagram
+    class DetectionWindows {
+        <<abstract>>
+    }
+
+    class DriftDetector
+
+    class Trigger {
+        <<abstract>>
+        +void init_trigger(TriggerContext context)
+        +Generator[Triggers] inform(new_data)
+        +void inform_previous_model(int previous_model_id)
+    }
+
+    class TimeTrigger {
+    }
+
+    class DataAmountTrigger {
+    }
+
+    class DriftDetector {
+        <<abstract>>
+    }
+
+    class DataDriftTrigger {
+        +DataDriftTriggerConfig config
+        +DetectionWindows _windows
+        +dict[MetricId, DriftDecisionPolicy] decision_policies
+        +dict[MetricId, DriftDetector] distance_detectors
+
+        +void init_trigger(TriggerContext context)
+        +Generator[triggers] inform(new_data)
+        +void inform_previous_model(int previous_model_id)
+    }
+
+    class DriftDecisionPolicy {
+        <<abstract>>
+    }
+
+    DataDriftTrigger *-- "1" DataDriftTriggerConfig
+    DataDriftTrigger *-- "1" DetectionWindows
+    DataDriftTrigger *-- "|metrics|" DriftDetector
+    DataDriftTrigger *-- "|metrics|" DriftDecisionPolicy
+
+    Trigger <|-- DataDriftTrigger
+    Trigger <|-- TimeTrigger
+    Trigger <|-- DataAmountTrigger
+```
+
+### DetectionWindows Hierarchy
+
+The `DetectionWindows` class serves as the abstract base for specific windowing strategies like `AmountDetectionWindows` and `TimeDetectionWindows`. These classes are responsible for both storing the actual data windows for reference and current data and for defining a strategy for updating and managing these windows.
+
+```mermaid
+classDiagram
+    class DetectionWindows {
+        <<abstract>>
+        +Deque current
+        +Deque current_reservoir
+        +Deque reference
+        +void inform_data(list[tuple[int, int]])
+        +void inform_trigger()
+    }
+
+    class AmountDetectionWindows {
+        +AmountWindowingStrategy config
+        +void inform_data(list[tuple[int, int]])
+        +void inform_trigger()
+    }
+
+    class TimeDetectionWindows {
+        +TimeWindowingStrategy config
+        +void inform_data(list[tuple[int, int]])
+        +void inform_trigger()
+    }
+
+    DetectionWindows <|-- AmountDetectionWindows
+    DetectionWindows <|-- TimeDetectionWindows
+```
+
+### DriftDetector Hierarchy
+
+The `DriftDetector` class is an abstract base class for detectors like `AlibiDriftDetector` and `EvidentlyDriftDetector`, which use different metrics to measure the distance between the current and reference data distributions. It generates distance values.
```

Comment:
Maybe add one sentence here that the underlying packages these detectors rely on also support making decisions, we implement our own decision logic in our drift decision policy, and ignore the output from the detectors.